src/llmcompressor/entrypoints/utils.py

```diff
@@ -79,20 +84,28 @@ def post_process(
     Raises:
         ValueError: If saving fails due to an invalid `output_dir` or other issues.
     """
-    if output_dir is not None:
+    if model_args is not None and output_dir is not None:
+        if recipe_args is not None and getattr(recipe_args, "stage", None) is not None:
+            output_dir = os.path.join(output_dir, recipe_args.stage)
+            os.makedirs(output_dir, exist_ok=True)
+            logger.info(f"[Save] Stage detected. Updating output_dir to {output_dir}")
+
         model_args.model.save_pretrained(
-            output_dir,
-            save_compressed=model_args.save_compressed,
+            output_dir, save_compressed=model_args.save_compressed
         )
-        if model_args.processor:
+
+        if model_args.processor is not None:
             model_args.processor.save_pretrained(output_dir)
-        return
 
-    logger.warning(
-        "Optimized model is not saved. To save, please provide",
-        "`output_dir` as input arg.",
-        "Ex. `oneshot(..., output_dir=...)`",
-    )
+    else:
+        logger.warning(
+            "Optimized model is not saved. To save, please provide",
+            "`output_dir` as input arg.",
+            "Ex. `oneshot(..., output_dir=...)`",
+        )
+
+    # Reset the one-time-use session upon completion
+    reset_session()
```

Comment:
is this a big thing? if so, we may want to call it out in docstring as a side effect?

Reply:
There are places where the info of the session is used to retrieve info. 
We can reset once it finishes running - which i think should be default, but the above need to be addressed. Ill do that in the follow up pr. 
For now resets if the user wants to reset